src/Worker.Extensions.DurableTask/DurableTaskClientExtensions.cs

```diff
@@ -17,6 +18,52 @@ namespace Microsoft.Azure.Functions.Worker;
 /// </summary>
 public static class DurableTaskClientExtensions
 {
+    /// <summary>
+    /// Creates an HTTP response that is useful for checking the status of the specified instance.
+    /// </summary>
+    /// <param name="client">The <see cref="DurableTaskClient"/>.</param>
+    /// <param name="request">The HTTP request that this response is for.</param>
+    /// <param name="instanceId">The ID of the orchestration instance to check.</param>
+    /// <param name="cancellation">The cancellation token.</param>
+    /// <returns>An HTTP 202 response with a Location header and a payload containing instance control URLs.</returns>
+    public static Task<HttpResponseData> CreateCheckStatusResponseAsync(
+        this DurableTaskClient client,
+        HttpRequestData request,
+        string instanceId,
+        CancellationToken cancellation = default)
+    {
+        return client.CreateCheckStatusResponseAsync(request, instanceId, HttpStatusCode.Accepted, cancellation);
+    }
+
+    /// <summary>
+    /// Creates an HTTP response that is useful for checking the status of the specified instance.
+    /// </summary>
+    /// <param name="client">The <see cref="DurableTaskClient"/>.</param>
+    /// <param name="request">The HTTP request that this response is for.</param>
+    /// <param name="instanceId">The ID of the orchestration instance to check.</param>
+    /// <param name="statusCode">The status code.</param>
+    /// <param name="cancellation">The cancellation token.</param>
+    /// <returns>An HTTP response with a Location header and a payload containing instance control URLs.</returns>
+    public static async Task<HttpResponseData> CreateCheckStatusResponseAsync(
+        this DurableTaskClient client,
+        HttpRequestData request,
+        string instanceId,
+        HttpStatusCode statusCode,
+        CancellationToken cancellation = default)
+    {
+        if (client is null)
+        {
+            throw new ArgumentNullException(nameof(client));
+        }
+
+        HttpResponseData response = request.CreateResponse(statusCode);
+        object payload = SetHeadersAndGetPayload(client, request, response, instanceId);
+
+        ObjectSerializer serializer = GetObjectSerializer(response);
+        await serializer.SerializeAsync(response.Body, payload, payload.GetType(), cancellation);
```

Comment:
so just to make sure I understand - the main blocker _before_ was that we were calling a synchronous serialization method?

Reply:
Yup! AspNetCore by default will throw if you try to use sync APIs on the response body.